### PR TITLE
Speed up job runs page

### PR DESCRIPTION
**Pull request by johnnymugs**
- Add GIN index for searching across multiple fields
- Reduce 2 queries down to 1 by including count

### CODE REVIEW

File: explorer/src/queries/search.ts [hunk 1]
```diff
@@ -23,20 +43,13 @@ const searchBuilder = (searchQuery?: string): SelectQueryBuilder<JobRun> => {
   let query = getRepository(JobRun).createQueryBuilder('job_run')
 
   if (searchQuery != null) {
-    const searchTokens = searchQuery.split(/\s+/)
-    const normalizedSearchTokens = searchTokens.map(normalizeSearchToken)
+    let searchTokens = searchQuery.split(/\s+/)
```
Comment (j16r) [inline explorer/src/queries/search.ts:46]: It'd be nice if these two lines were put into their own function to make it a bit clearer, then searchTokens can be `const` as well for the duration of this function.

File: explorer/src/queries/search.ts [hunk 2]
```diff
@@ -23,20 +43,13 @@ const searchBuilder = (searchQuery?: string): SelectQueryBuilder<JobRun> => {
...
-      .orWhere('job_run.requester IN(:...normalizedSearchTokens)', {
-        normalizedSearchTokens,
-      })
-      .orWhere('job_run.requestId IN(:...normalizedSearchTokens)', {
-        normalizedSearchTokens,
-      })
-      .orWhere('job_run.txHash IN(:...normalizedSearchTokens)', {
-        normalizedSearchTokens,
-      })
+      .addSelect('COUNT(1) OVER() AS "totalRecords"')
```
Comment (j16r) [inline explorer/src/queries/search.ts:49]: Would `COUNT(*)` be better? (I haven't seen (1) before, what does it do?)

**Comment by rupurt** [comment]
@johnnymugs can you also post the query plan before/after this change?

**Comment by johnnymugs** [comment]
```
explorer_dev=# SET enable_seqscan=OFF;
SET
```
Old query that returns 170k rows
```
explorer_dev=# EXPLAIN ANALYZE SELECT COUNT(job_run.id)
FROM job_run
WHERE job_run."runId" IN ('0x1597F8CddA5Bc5E579BAdC4d1173d28CbAbD1339')
  OR job_run."jobId" IN ('0x1597F8CddA5Bc5E579BAdC4d1173d28CbAbD1339')
  OR job_run."requester" IN ('0x1597F8CddA5Bc5E579BAdC4d1173d28CbAbD1339')
  OR job_run."requestId" IN ('0x1597F8CddA5Bc5E579BAdC4d1173d28CbAbD1339')
  OR job_run."txHash" IN ('0x1597F8CddA5Bc5E579BAdC4d1173d28CbAbD1339');

                           QUERY PLAN
---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
 Finalize Aggregate  (cost=68487.53..68487.54 rows=1 width=8) (actual time=664.998..664.998 rows=1 loops=1)
   ->  Gather  (cost=68487.32..68487.53 rows=2 width=8) (actual time=664.823..668.624 rows=3 loops=1)
         Workers Planned: 2
         Workers Launched: 2
         ->  Partial Aggregate  (cost=67487.32..67487.33 rows=1 width=8) (actual time=662.968..662.968 rows=1 loops=3)
               ->  Parallel Bitmap Heap Scan on job_run  (cost=36339.63..67306.48 rows=72337 width=37) (actual time=618.566..658.247 rows=57704 loops=3)
                     Recheck Cond: (("runId" = '0x1597F8CddA5Bc5E579BAdC4d1173d28CbAbD1339'::citext) OR ("jobId" = '0x1597F8CddA5Bc5E579BAdC4d1173d28CbAbD1339'::citext) OR (requester = '0x1597F8CddA5Bc5E579BAdC4d1173d28CbAbD1339'::citext) OR ("requestId" = '0x1597F8CddA5Bc5E579BAdC4d1173d28CbAbD1339'::citext) OR ("txHash" = '0x1597F8CddA5Bc5E579BAdC4d1173d28CbAbD1339'::citext))
                     Heap Blocks: exact=5840
                     ->  BitmapOr  (cost=36339.63..36339.63 rows=173637 width=0) (actual time=617.721..617.721 rows=0 loops=1)
                           ->  Bitmap Index Scan on job_run_chainlink_node_id_run_id_idx  (cost=0.00..21522.42 rows=1 width=0) (actual time=288.698..288.698 rows=0 loops=1)
                                 Index Cond: ("runId" = '0x1597F8CddA5Bc5E579BAdC4d1173d28CbAbD1339'::citext)
                           ->  Bitmap Index Scan on job_run_job_id_idx  (cost=0.00..4.50 rows=10 width=0) (actual time=0.037..0.037 rows=0 loops=1)
                                 Index Cond: ("jobId" = '0x1597F8CddA5Bc5E579BAdC4d1173d28CbAbD1339'::citext)
                           ->  Bitmap Index Scan on job_run_requester_idx  (cost=0.00..14586.41 rows=173599 width=0) (actual time=328.886..328.886 rows=173113 loops=1)
                                 Index Cond: (requester = '0x1597F8CddA5Bc5E579BAdC4d1173d28CbAbD1339'::citext)
                           ->  Bitmap Index Scan on job_run_request_id_idx  (cost=0.00..4.68 rows=18 width=0) (actual time=0.053..0.053 rows=0 loops=1)
                                 Index Cond: ("requestId" = '0x1597F8CddA5Bc5E579BAdC4d1173d28CbAbD1339'::citext)
                           ->  Bitmap Index Scan on job_run_tx_hash_idx  (cost=0.00..4.61 rows=9 width=0) (actual time=0.044..0.044 rows=0 loops=1)
                                 Index Cond: ("txHash" = '0x1597F8CddA5Bc5E579BAdC4d1173d28CbAbD1339'::citext)
 Planning Time: 0.126 ms
 Execution Time: 668.682 ms
(21 rows)
```
Old query that returns 0 rows
```
explorer_dev=# EXPLAIN ANALYZE SELECT COUNT(job_run.id)
FROM job_run
WHERE job_run."runId" IN ('0x1597F8CddA5Bc5E579BAdC4d1173d28CbAbD1336')
  OR job_run."jobId" IN ('0x1597F8CddA5Bc5E579BAdC4d1173d28CbAbD1336')
  OR job_run."requester" IN ('0x1597F8CddA5Bc5E579BAdC4d1173d28CbAbD1336')
  OR job_run."requestId" IN ('0x1597F8CddA5Bc5E579BAdC4d1173d28CbAbD1336')
  OR job_run."txHash" IN ('0x1597F8CddA5Bc5E579BAdC4d1173d28CbAbD1336');
Time: 261.396 ms
                     QUERY PLAN
---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
 Aggregate  (cost=21716.58..21716.59 rows=1 width=8) (actual time=260.687..260.688 rows=1 loops=1)
   ->  Bitmap Heap Scan on job_run  (cost=21540.74..21716.47 rows=45 width=37) (actual time=260.681..260.681 rows=0 loops=1)
         Recheck Cond: (("runId" = '0x1597F8CddA5Bc5E579BAdC4d1173d28CbAbD1336'::citext) OR ("jobId" = '0x1597F8CddA5Bc5E579BAdC4d1173d28CbAbD1336'::citext) OR (requester = '0x1597F8CddA5Bc5E579BAdC4d1173d28CbAbD1336'::citext) OR ("requestId" = '0x1597F8CddA5Bc5E579BAdC4d1173d28CbAbD1336'::citext) OR ("txHash" = '0x1597F8CddA5Bc5E579BAdC4d1173d28CbAbD1336'::citext))
         ->  BitmapOr  (cost=21540.74..21540.74 rows=45 width=0) (actual time=260.678..260.678 rows=0 loops=1)
               ->  Bitmap Index Scan on job_run_chainlink_node_id_run_id_idx  (cost=0.00..21522.42 rows=1 width=0) (actual time=260.476..260.476 rows=0 loops=1)
                     Index Cond: ("runId" = '0x1597F8CddA5Bc5E579BAdC4d1173d28CbAbD1336'::citext)
               ->  Bitmap Index Scan on job_run_job_id_idx  (cost=0.00..4.50 rows=10 width=0) (actual time=0.048..0.048 rows=0 loops=1)
                     Index Cond: ("jobId" = '0x1597F8CddA5Bc5E579BAdC4d1173d28CbAbD1336'::citext)
               ->  Bitmap Index Scan on job_run_requester_idx  (cost=0.00..4.48 rows=8 width=0) (actual time=0.059..0.059 rows=0 loops=1)
                     Index Cond: (requester = '0x1597F8CddA5Bc5E579BAdC4d1173d28CbAbD1336'::citext)
               ->  Bitmap Index Scan on job_run_request_id_idx  (cost=0.00..4.68 rows=18 width=0) (actual time=0.049..0.049 rows=0 loops=1)
                     Index Cond: ("requestId" = '0x1597F8CddA5Bc5E579BAdC4d1173d28CbAbD1336'::citext)
               ->  Bitmap Index Scan on job_run_tx_hash_idx  (cost=0.00..4.61 rows=9 width=0) (actual time=0.043..0.043 rows=0 loops=1)
                     Index Cond: ("txHash" = '0x1597F8CddA5Bc5E579BAdC4d1173d28CbAbD1336'::citext)
 Planning Time: 0.128 ms
 Execution Time: 260.743 ms
(16 rows)
```
New query (170k+ rows)
```
explorer_dev=# EXPLAIN ANALYZE SELECT COUNT(job_run.id) FROM job_run WHERE ARRAY["runId","jobId","requestId","requester","txHash"] @> ARRAY['0x1597F8CddA5Bc5E579BAdC4d1173d28CbAbD1339']::citext[];
                                                                    QUERY PLAN
---------------------------------------------------------------------------------------------------------------------------------------------------
 Aggregate  (cost=3862.33..3862.34 rows=1 width=8) (actual time=74.432..74.432 rows=1 loops=1)
   ->  Bitmap Heap Scan on job_run  (cost=24.83..3859.48 rows=1140 width=37) (actual time=19.056..61.100 rows=173113 loops=1)
         Recheck Cond: (ARRAY["runId", "jobId", "requestId", requester, "txHash"] @> '{0x1597F8CddA5Bc5E579BAdC4d1173d28CbAbD1339}'::citext[])
         Heap Blocks: exact=14427
         ->  Bitmap Index Scan on updog_ary_idx  (cost=0.00..24.55 rows=1140 width=0) (actual time=17.136..17.136 rows=173113 loops=1)
               Index Cond: (ARRAY["runId", "jobId", "requestId", requester, "txHash"] @> '{0x1597F8CddA5Bc5E579BAdC4d1173d28CbAbD1339}'::citext[])
 Planning Time: 1.492 ms
 Execution Time: 74.528 ms
(8 rows)

Time: 76.486 ms
```

New query (0 rows)
```
explorer_dev=# EXPLAIN ANALYZE SELECT COUNT(job_run.id) FROM job_run WHERE ARRAY["runId","jobId","requestId","requester","txHash"] @> ARRAY['0x1597F8CddA5Bc5E579BAdC4d1173d28CbAbD1336']::citext[];
                                                                    QUERY PLAN
---------------------------------------------------------------------------------------------------------------------------------------------------
 Aggregate  (cost=3862.33..3862.34 rows=1 width=8) (actual time=0.041..0.041 rows=1 loops=1)
   ->  Bitmap Heap Scan on job_run  (cost=24.83..3859.48 rows=1140 width=37) (actual time=0.039..0.039 rows=0 loops=1)
         Recheck Cond: (ARRAY["runId", "jobId", "requestId", requester, "txHash"] @> '{0x1597F8CddA5Bc5E579BAdC4d1173d28CbAbD1336}'::citext[])
         ->  Bitmap Index Scan on updog_ary_idx  (cost=0.00..24.55 rows=1140 width=0) (actual time=0.037..0.037 rows=0 loops=1)
               Index Cond: (ARRAY["runId", "jobId", "requestId", requester, "txHash"] @> '{0x1597F8CddA5Bc5E579BAdC4d1173d28CbAbD1336}'::citext[])
 Planning Time: 0.089 ms
 Execution Time: 0.063 ms
(7 rows)

Time: 0.445 ms
```

File: explorer/src/queries/search.ts (hunk 2 above)
Comment (johnnymugs) [reply explorer/src/queries/search.ts:49]: Ah, they are equivalent but it's good to stay consistent across the project so I'll update this.

**Pull request merged**